tap-agent/src/agent/sender_allocation.rs

```diff
@@ -466,92 +466,132 @@ impl SenderAllocationState {
-                 This may happen if your `rav_request_trigger_value` is too low \
-                 and no receipts were found outside the `rav_request_timestamp_buffer_ms`.\
-                 You can fix this by increasing the `rav_request_trigger_value`."
-                ),
-                _ => e.into(),
-            })?;
-        let valid_receipts: Vec<_> = valid_receipts
-            .into_iter()
-            .map(|r| r.signed_receipt().clone())
-            .collect();
-        let rav_response_time_start = Instant::now();
-        let response: JsonRpcResponse<EIP712SignedMessage<ReceiptAggregateVoucher>> = self
-            .http_client
-            .request(
-                "aggregate_receipts",
-                rpc_params!(
-                    "0.0", // TODO: Set the version in a smarter place.
-                    valid_receipts,
-                    previous_rav
-                ),
-            )
             .await?;
-
-        let rav_response_time = rav_response_time_start.elapsed();
-        RAV_RESPONSE_TIME
-            .with_label_values(&[&self.sender.to_string()])
-            .observe(rav_response_time.as_secs_f64());
-
-        // we only save invalid receipts when we are about to store our rav
-        //
-        // store them before we call remove_obsolete_receipts()
-        if !invalid_receipts.is_empty() {
-            warn!(
-                "Found {} invalid receipts for allocation {} and sender {}.",
-                invalid_receipts.len(),
-                self.allocation_id,
-                self.sender
-            );
-
-            // Save invalid receipts to the database for logs.
-            // TODO: consider doing that in a spawned task?
-            self.store_invalid_receipts(invalid_receipts.as_slice())
+        match (
+            expected_rav,
+            valid_receipts.is_empty(),
+            invalid_receipts.is_empty(),
+        ) {
+            // No valid receipts, but there are invalid
+            (Err(tap_core::Error::NoValidReceiptsForRAVRequest), true, false) => {
+                warn!(
+                    "Found {} invalid receipts for allocation {} and sender {}.",
+                    invalid_receipts.len(),
+                    self.allocation_id,
+                    self.sender
+                );
+                self.store_invalid_receipts(invalid_receipts.as_slice())
+                    .await?;
+                let min_timestamp = invalid_receipts
+                    .first()
+                    .expect("There should be atleast one receipt")
+                    .signed_receipt()
+                    .message
+                    .timestamp_ns;
+                let max_timestamp = invalid_receipts
+                    .last()
+                    .expect("There should be atleast one receipt")
+                    .signed_receipt()
+                    .message
+                    .timestamp_ns;
```

Comment:
Can you add a comment here about the fact that we expect the receipts to already be sorted by timestamp since we're using [`safe_truncate_receipts`](https://github.com/semiotic-ai/timeline-aggregation-protocol/blob/04b134a4e0268f5b3e87c49639e4d8315b59017a/tap_core/src/manager/adapters/receipt.rs#L102) which happens to do the sort for us [here](https://github.com/graphprotocol/indexer-rs/blob/20bae14e5bfb8e963c444c6dcd475ff0332155d1/tap-agent/src/tap/context/receipt.rs#L157).

Reply:
And also [`collect_receipts`](https://github.com/semiotic-ai/timeline-aggregation-protocol/blob/04b134a4e0268f5b3e87c49639e4d8315b59017a/tap_core/src/manager/tap_manager.rs#L100) doesn't change the order of receipts when checking / moving receipts to the invalid receipts list.